osutil/sync_dir_test.go

```diff
@@ -0,0 +1,175 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/osutil"
+)
+
+type EnsureDirStateSuite struct {
+	dir  string
+	glob string
+}
+
+var _ = Suite(&EnsureDirStateSuite{glob: "*.snap"})
+
+func (s *EnsureDirStateSuite) SetUpTest(c *C) {
+	s.dir = c.MkDir()
+}
+
+func (s *EnsureDirStateSuite) TestVerifiesExpectedFiles(c *C) {
+	name := filepath.Join(s.dir, "expected.snap")
+	err := ioutil.WriteFile(name, []byte(`expected`), 0600)
+	c.Assert(err, IsNil)
+	changed, removed, err := osutil.EnsureDirState(s.dir, s.glob, map[string]*osutil.FileState{
+		"expected.snap": {Content: []byte(`expected`), Mode: 0600},
+	})
+	c.Assert(err, IsNil)
+	// Report says that nothing has changed
+	c.Assert(changed, HasLen, 0)
+	c.Assert(removed, HasLen, 0)
+	// The content is correct
+	content, err := ioutil.ReadFile(path.Join(s.dir, "expected.snap"))
+	c.Assert(err, IsNil)
+	c.Assert(content, DeepEquals, []byte(`expected`))
+	// The permissions are correct
+	stat, err := os.Stat(name)
+	c.Assert(err, IsNil)
+	c.Assert(stat.Mode().Perm(), Equals, os.FileMode(0600))
+}
```

Comment:
Disagree a bit. Such trivial constants in place make the test more readable compared to a variable we need to look for elsewhere.

An exception is when we have logic that is interdependent but not obviously so. For example, we have two loops that need to have the same number of iterations.

We should fix the inconsistent use of quotes, though ("expected").